desktop/core/src/desktop/api_public.py

```diff
@@ -52,6 +53,10 @@ def get_context_namespaces(request, interface):
   django_request = get_django_request(request)
   return desktop_api.get_context_namespaces(django_request, interface)
 
+@api_view(["GET"])
+@permission_classes([AllowAny])
```

Comment:
Yup, we also display it on the login page.

Reply:
@bjornalm Then this is specific to Hue UI, if are are not authenticating here, then anyone can use this public API.
Since Hue UI needs this and for that Hue user need not be logged in, then the fix can be at core API implementation (which is private API perspective).

We can try adding `@login_notrequired` decorator here.
https://github.com/cloudera/hue/pull/3117/files#diff-6bfeceac61e0c89f79fb2abdb28866f5a4cb21bee453a30bd376012d141c3c51R87

Reply:
There's no harm in anyone else using this endpoint. And the idea is for the Hue UI to also only use the public API. @login_notrequired doesn't affect the public endpoints with django rest.